### ISSUE TITLE
Error: failed to generate ledger key:

### ISSUE THREAD

**Issue by Subfortytwo** [crypto-org-chain/chain-main#751]
**Describe the bug**
When trying to generate a keyfile from ledger, I am getting the following error:

```
$ ~/go/bin/chain-maind keys add ledger --ledger --keyring-backend file
Error: failed to generate ledger key: failed to recover pubkey: [APDU_CODE_DATA_INVALID] Referenced data reversibly blocked (invalidated): address rejected for path m/44'/394'/0'/0/0
Usage:
  chain-maind keys add <name> [flags]

Flags:
[...]
```
I tested with both the github repository and binary.
Thanks for your help.

**Comment by tomtau** [comment]
@Subfortytwo are you using the Crypto.org Chain Ledger app?

**Comment by Subfortytwo** [comment]
No, I am using Keplr to stake my CRO, so I have to use the Cosmos ledger app.
Basically what I am trying to do is be able to autocompound with restake: https://restake.app/cosmoshub.
This means sending the following two commands to the delegator:

```
/cosmos.distribution.v1beta1.MsgWithdrawDelegatorReward 
/cosmos.staking.v1beta1.MsgDelegate

```

Maybe do you have a workaround (without having to undelegate to move the funds)?

Thank you very much.

**Comment by tomtau** [comment]
if you're using Keplr and the Cosmos Ledger app, then you may need to add the extra `--coin-type 118` argument, as the default one of Crypto.org Chain (394) won't work on the Cosmos (118) Ledger app.
